README.md

```diff
@@ -33,33 +33,97 @@ Examples:
     ~foreman/config/hooks/smart_proxy/after_create/01_email_operations.sh
     ~foreman/config/hooks/audited/adapters/active_record/audit/after_create/01_syslog.sh
 
-## SELinux notes
-
-When using official installation on Red Hat and Fedora system, note that
-SELinux is turned on by default and Foreman is running in confined mode. Make
-sure that hook scripts has the correct context (`foreman_hook_t` on
-RHEL7+/Fedora 19+ or `bin_t` on RHEL6):
-
-    restorecon -RvF /usr/share/foreman/config/hooks
-
-Also keep in mind that the script is running confined, therefore some actions
-might be denied by SELinux. Check audit.log and use audit2allow and other
-tools when writing scripts.
-
 ## Objects / Models
 
 Every object (or model in Rails terms) in Foreman can have hooks.  Check
-`~foreman/app/models` for the full list, but these are the interesting ones:
-
-* `host/managed` (or `host` in Foreman 1.1)
+`~foreman/app/models` for the full list, but these are the interesting ones
+from Foreman 1.11 Core or some its plugins:
+
+* `architecture`
+* `bookmark`
+* `compute_resource`
+* `config_report`
+* `container`
+* `domain`
+* `environment`
+* `fact_name`
+* `feature`
+* `filter`
+* `filtering`
+* `foreman_docker/docker`
+* `foreman_openscap/arf_report`
+* `foreman_openscap/compliance_status`
+* `foreman_tasks/recurring_logic`
+* `foreman_tasks/task`
+* `host/base`
+* `host/managed`
+* `host/discovered`
+* `host_status/build_status`
+* `host_status/configuration_status`
+* `host_status/execution_status`
+* `host_status/status`
+* `hostgroup`
+* `job_template`
+* `katello/errata_status`
+* `katello/erratum`
+* `katello/host/content_facet`
+* `katello/host/subscription_facet`
+* `katello/model`
+* `katello/puppet_module`
+* `katello/repository`
+* `katello/rhsm_fact_name`
+* `katello/rpm`
+* `katello/subscription_status`
+* `location`
+* `log`
+* `medium`
+* `model`
+* `nic/base`
+* `nic/bmc`
+* `nic/bond`
+* `nic/bridge`
+* `nic/interface`
+* `nic/managed`
+* `operatingsystem`
+* `organization`
+* `permission`
+* `provisioning_template`
+* `ptable`
+* `puppetclass`
+* `realm`
+* `redhat`
+* `remote_execution_feature`
 * `report`
+* `role`
+* `setting`
+* `setting/auth`
+* `setting/bootdisk`
+* `setting/discovered`
+* `setting/foreman_tasks`
+* `setting/general`
+* `setting/katello`
+* `setting/provisioning`
+* `setting/puppet`
+* `setting/remote_execution`
+* `smart_proxy`
+* `subnet`
+* `taxonomy`
+* `template`
+* `user`
+* `user_role`
+* `usergroup`
+
+To generate list of *all* possible models, issue the following command:
+
+    # foreman-rake console
+    > ActiveRecord::Base.descendants.collect(&:name).collect(&:underscore).sort
```

Comment:
I used this command to generate the list. I think it won't hurt us to explicitly list it all here.